canonical-web-and-design/maas-ui
# Dismiss notifications with x icon

## Done

- Update the notification list to dismiss with the x icon.
- Convert component to TypeScript.
- Update to use Notification component.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)

### QA steps

- Add some notifications (I use `make harness` and then something like `from maasserver.testing.factory import factory` then `for i in range(0, 10): factory.make_Notification(admins=True)`).
- Load a React page.
- The notifications should each have an x icon.
- Clicking the icon should dismiss the notification.

## Fixes

Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/2060.
